### ISSUE TITLE
Text Chat

### ISSUE THREAD

**Issue by mikeal** [mikeal/peer-call#11]
A simple text based chat interface would be great!

Maybe this should use something like hypercore https://github.com/mafintosh/hypercore for simple append-only synchronization. We could use simple signing (already implemented in the crypto underneath killa-beez) to verify that each message was from each person in the swarm.

**Comment by mafintosh** [comment]
hypercore is also signed so you just need to sign the hypercore feed key to get this working

**Comment by mikeal** [comment]
Yup, I've got a local branch of signal-exchange to migrate to sodium for the encryption so that we can just plug right into hypercore, I'm just blocked on figuring out how to do both signing and encryption with same keypair (mathias and i have an ongoing conversation on IRC about how to accomplish this).

**Comment by thezachbruno** [comment]
Hello everyone!
I know this is probably not the best place for a newbie open source contributor to start, but is there anything I can do to help contribute to this project? Any simple tasks that no one feels like doing I would be in for. I think this project is one beautiful concept and I would love to be a part of it while learning as much as possible.

**Comment by mikeal** [comment]
@chemeyo this _specific_ issue probably isn't the best first issue to tackle, but there are plenty of other good first issues to go after. The best thing for you to do is probably to log a new issue and ask for "good first fixes" and the contributors can use it to suggest things.

Also, on a personal note, I love the One Punch Man reference in your avatar :)

![wiggle-eyebrows](https://cloud.githubusercontent.com/assets/579/19425329/015aadd0-93e6-11e6-8dbb-49e5e1d27907.gif)

**Comment by thezachbruno** [comment]
Hey man that is perfect thank you! I just submitted the new issue now! I appreciate it I know it's probably annoying having me in here but I have to start somewhere -- and go OPM :D cannot go wrong!!

**Comment by suamorales** [comment]
what about websockets. would something like socket.io work?

**Comment by suamorales** [comment]
bump @mikeal

**Comment by mikeal** [comment]
@suamorales I've started working on this https://github.com/mikeal/bong-bong

websockets/socket.io assume that there is infrastructure in place, which we have been able to avoid by using pure p2p technologies like WebRTC. This new project, `bong-bong`, uses the same underlying library for p2p we use already in `roll-call` so it'll be a good fit to slot in once it's ready.

**Comment by mikeal** [comment]
I'm going back to some of my earlier thinking on this.

The main mode of communication in Roll Call is audio. If people need to send text back and forth I'm interested in ideas that make that realtime but secondary to the audio experience.

A traditional text chat is expensive in terms of UI and data sync and I'm not entirely sure it's the right direction to go in.
